.claude/agents/project-structure-validator/build-component-graph.sh

```
@@ -0,0 +1,41 @@
+OUTPUT="$SCRIPT_DIR/.component-graph.json"
+
+echo "{" > "$OUTPUT"
+echo '  "components": {' >> "$OUTPUT"
+
+first=true
+find "$DIR" -name "*.tsx" -type f ! -path "*/node_modules/*" ! -name "*.test.tsx" ! -name "*.stories.tsx" | while read file; do
+  component=$(basename "$file" .tsx)
+  path="${file#$DIR/}"
+
+  # Count imports
+  count=$(grep -r "from.*['\"].*$component['\"]" "$DIR" --include="*.tsx" --include="*.ts" 2>/dev/null | grep -v "$file" | wc -l | tr -d ' ')
```

Shellcheck (0.11.0)

[error] Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).

(SC1087)